Source/WebKit/WebKitSwift/WritingTools/IntelligenceTextEffectViewManager.swift

```diff
@@ -0,0 +1,195 @@
+        if self.suppressEffectView {
+            effectView.isHidden = true
+        }
+
+        // UIKit expects subviews of the effect view to be added after the effect view is added to its parent.
+        effectView.initializeSubviews()
+
+        self.effectView = effectView
+    }
+
+    private func destroyViewIfNeeded() {
+        guard self.activePonderingEffect == nil && self.activeReplacementEffect == nil else {
+            return
+        }
+
+        self.effectView?.removeFromSuperview()
+        self.effectView = nil
+    }
+
+    func setActivePonderingEffect(_ effect: PlatformIntelligencePonderingTextEffect<IntelligenceTextEffectChunk>?) async {
+        guard (self.activePonderingEffect == nil && effect != nil) || (self.activePonderingEffect != nil && effect == nil) else {
+            return
+        }
+
+        let oldEffect = self.activePonderingEffect
+        self.activePonderingEffect = effect
+
+        if let effect {
+            self.setupViewIfNeeded()
+            await self.effectView?.addEffect(effect)
+        } else {
+            // This needs to be manually invoked rather than relying on the associated delegate method being called. This is
+            // because when removing an effect, the delegate method is invoked after the effect removal function ends. Invoking
+            // this method manually ensures that the replacement effect doesn't start until the visibility has actually changed
+            // and this function terminates.
+            //
+            // Therefore, the delegate method itself must avoid any work so that it can be synchronous, which is what the platform
+            // interfaces expect.
+            await self.source?.updateTextChunkVisibility(oldEffect!.chunk, visible: true)
+
+            await self.effectView?.removeEffect(oldEffect!.id)
+
+            self.destroyViewIfNeeded()
+        }
+    }
+
+    func setActiveReplacementEffect(_ effect: PlatformIntelligenceReplacementTextEffect<IntelligenceTextEffectChunk>?) async {
+        guard (self.activeReplacementEffect == nil && effect != nil) || (self.activeReplacementEffect != nil && effect == nil) else {
+            assertionFailure("Intelligence text effect coordinator: trying to either set a new replacement effect when there is an ongoing one, or trying to remove an effect when there are none.")
+            return
+        }
+
+        let oldEffect = self.activeReplacementEffect
+        self.activeReplacementEffect = effect
+
+        if let effect {
+            self.setupViewIfNeeded()
+            await self.effectView?.addEffect(effect)
+        } else {
+            await self.effectView?.removeEffect(oldEffect!.id)
+            self.destroyViewIfNeeded()
+        }
+    }
+
+    func hideEffects() async {
+        guard !self.suppressEffectView else {
+            return
+        }
+
+        // On macOS, the effect view is incapable of scrolling with the web view. Therefore, the effects need to be suppressed
```

Comment:
This is not categorically true (we have a scrolling tree and can move things around in the UI process), we just haven't put in the effort. "Currently doesn't scroll with the web content" (the web view doesn't scroll) might be more accurate.